pkg/controllers/resources/pods/translate/sa_token_secret.go

```diff
@@ -0,0 +1,144 @@
+package translate
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/loft-sh/vcluster/pkg/util/translate"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	PodKind string = "Pod"
+)
+
+var PodServiceAccountTokenSecretName string
+
+func SecretNameFromPodName(podName, namespace string) string {
+	return translate.Default.PhysicalName(fmt.Sprintf("%s-sa-token", podName), namespace)
+}
+
+func GetSecretIfExists(ctx context.Context, pClient client.Client, vPodName, vNamespace string) (*corev1.Secret, bool, error) {
+	secret := &corev1.Secret{}
+
+	err := pClient.Get(ctx, types.NamespacedName{
+		Name:      SecretNameFromPodName(vPodName, vNamespace),
+		Namespace: translate.Default.PhysicalNamespace(vNamespace),
+	}, secret)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return nil, false, nil
+		}
+
+		return nil, false, err
+	}
+
+	return secret, true, nil
+}
+
+func SATokenSecret(ctx context.Context, pClient client.Client, vPod *corev1.Pod, tokens map[string]string) error {
+	_, exists, err := GetSecretIfExists(ctx, pClient, vPod.Name, vPod.Namespace)
+	if err != nil {
+		return err
+	}
+
+	if !exists {
+		// create to secret with the given token
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      SecretNameFromPodName(vPod.Name, vPod.Namespace),
+				Namespace: translate.Default.PhysicalNamespace(vPod.Namespace),
+
+				Annotations: map[string]string{
+					translate.SkipBacksyncInMultiNamespaceMode: "true",
+				},
+			},
+			Type:       corev1.SecretTypeOpaque,
+			StringData: tokens,
+		}
+
+		if translate.Owner != nil {
+			typeAccessor, err := meta.TypeAccessor(translate.Owner)
+			if err != nil {
+				return err
+			}
+			secret.SetOwnerReferences([]metav1.OwnerReference{
+				{
+					APIVersion: typeAccessor.GetAPIVersion(),
+					Kind:       typeAccessor.GetKind(),
+					Name:       translate.Owner.GetName(),
+					UID:        translate.Owner.GetUID(),
+				},
+			})
```

Comment:
nit-pick: can be done with the util function [here](https://github.com/loft-sh/vcluster/blob/29acb7ccdb845fd5f6c7738a5eb9aacc1fa5cd2f/pkg/util/translate/translate.go#L41)

```suggestion
			secret.SetOwnerReferences(translate.GetOwnerReference(nil))
```